articles/coding-on-purpose.md

```diff
@@ -0,0 +1,99 @@
+
+> When we build software, our success is measured by whether the software can make the impact for which the customer is ready paying a certain amount of money.
+
+You don't need to be a millenial to understand this. This was true long before millenials were even born.
+
+## Purpose in Software Development Methodologies
+
+As long as there has been a professional software industry people in this industry have invented methods for approaching software development. There's one theme that shows up again and again in all methods: *managing purpose*.
+
+* In *RUP*, one tries to establish traceability from lower level items to higher level goals
+* If you work with *Use Cases* each Use Case represents a goal on a certain abstraction level and you can show how lower level Use Cases support higher level Use Cases in a use case diagram.
+* *User Stories* are usually formulated according to a template that also includes a *goal* and a *benefit* (e.t. "as a &lt;role&gt; I want to &lt;goal&gt; so that &lt;benefit&gt;").
+* With a *User Story Map* we can create a hierarchy of these *User Stories*.
+* An *Impact Map* traces down the high level goals to actors, impacts and finally deliverables
+* *Design Thinking* has a strong focus on *identifying and understanding customer needs*
+* The idea of the *Minimal Viable Product* makes the point that we should only build as much as is necessary for fulfilling the desired goal.
+
+And even beyond software develpment, the *why* topic seems to be very popular. One of the most famous representatives of the modern *why* movement is probably [Simon Sinek who tells us](https://startwithwhy.com/): *Start with Why!*
+
+## A Vacuum Of Purpose
+
+Let's go back to software again. Even though there's a lot of talk about impact, purpose and so on, there are also many areas with a vacuum of purpose. Some examples:
+
+* Code, which is mostly imperative in it's nature, does just declare *how* something is achieved but it does not state *why*. So while the purpose was hopefully clear to the one writing the code, it is likely to be unknown to the poor coder maintaining the code years later.
+* I've seen too many user stories that either don't state a goal and benefit or if they do, it's just a wild guess by somebody who wanted to satisfy the template structure. And worse: higher level items like *epics* miss the *purpose* even more often.
+* BDD (Behaviour Driven Design) typically uses the *Given, When, Then* pattern which only describes what the software does, but not why.
+* Many consulting companies get told *what* to build instead of *what problem to solve* or *what outcome to achieve*.
+* Inexperienced managers might tell you *what* to build instead of *what outcome to achieve*.

```

Comment:
* I like the examples - i would add "Estimations" and "Story Points" - these do not help to choose the right thing to implement - often I have seen that based on a Esitmation or "Story Points" the solution which cost the less was choosen, although one of the other solutions could have fitted its Purpose much more and have a much better impact.